src/server/cluster/cluster_config.cc

```diff
@@ -298,6 +298,18 @@ shared_ptr<ClusterConfig> ClusterConfig::CreateFromConfig(string_view my_id,
   return CreateFromConfig(my_id, config.value());
 }
 
+std::shared_ptr<ClusterConfig> ClusterConfig::CloneAndUpdate(const std::vector<SlotRange>& slots,
```

Comment:
we need to update ClusterShards config_; here as well otherwise we will not reply with the relevant moved error

Reply:
yes I know, but if we use the new approach that we discuss we don;t need these changes